EventGenerator/src/StoppedParticleReactionGun_module.cc

```diff
@@ -136,7 +141,12 @@ namespace mu2e {
       _hPdgId  = tfs->make<TH1F>("hPdgId" , "PDG ID"      ,  500,  -250, 250);
```

Comment:
In the early days we often made histograms/TTrees within producer modules.  After some experience we decided that, if all of the information being added to the histograms/TTrees was available in the data products made by the module, then we recommend removing the histograms/TTrees from the producer module and making a separate analyzer module to make the histograms/TTrees.   We have started to move existing histograms/TTrees to separate modules as targets of opportunity.  I would like you to do that here.

The only major exception to this is if there is transient information within the module that never makes it into a data product and you need to monitor the transient information.  In that case be sure that you can complete disable creating and filling the histograms with a fcl parameter.

Reply:
Hi Rob, We wanted the generated momentum spectra pre any filtering and reco conditions. Its the same information whcih is stored in the TH1F's which come out here but storing a TTree means its portable to python for anlaysis

Reply:
I looked at the module.  The histograms come *after* the event.put.  So if you move them to an analyzer module and add that module to one of the end paths they will have exactly the same content that they currently have.  Analyzer modules see every event, even the ones that fail some or all of the filters.  If you want an analyzer module to see events that pass a certain filter, or a combination of filters, you can add a SelectEvents parameter to the analyzer's parameter set to choose what events to see; art looks after the selection.